File: app/pkg/handlers/mappers/space.mapper.go
package mappers

import (
	"github.com/Serbroda/ragbag/app/gen"
	"github.com/Serbroda/ragbag/app/gen/restricted"
	"github.com/Serbroda/ragbag/app/pkg/utils"
)

func MapSpace(space gen.Space) restricted.SpaceDto {
	return restricted.SpaceDto{
		Description: utils.NullStringToString(space.Description),
		Id:          &space.ShortID,
		Name:        &space.Name,
		OwnerId:     &space.OwnerID,
		Visibility:  (*restricted.SpaceVisibility)(&space.Visibility),
	}
}

func MapSpaces(spaces []gen.Space) []restricted.SpaceDto {
	var dtos []restricted.SpaceDto
	for _, s := range spaces {
		dtos = append(dtos, MapSpace(s))
	}
	return dtos
}

File: ragbag-webapp/src/main/java/de/serbroda/ragbag/mappers/SpaceMapper.java
package de.serbroda.ragbag.mappers;

import de.serbroda.ragbag.dtos.space.CreateSpaceDto;
import de.serbroda.ragbag.dtos.space.SpaceDto;
import de.serbroda.ragbag.models.Space;
import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

@Mapper
public interface SpaceMapper {

    SpaceMapper INSTANCE = Mappers.getMapper(SpaceMapper.class);

    Space map(CreateSpaceDto source);

    SpaceDto map(Space source);
}
